File: algorythms/data_structure/linked_list.js
class LinkedListNode {
    constructor(value, next){
        this.value=value
        this.next=next
    }
}

const head = new LinkedListNode(5,
    new LinkedListNode(8,
        new LinkedListNode(12,
            new LinkedListNode(3, null)
        )
    )
)

class LinkedList {
    head = null

    add(value){
        this.head = new LinkedListNode(value, this.head)
    }

    insert(index, value){
        if(this.head === null)
            this.head = new LinkedListNode(value, null)
        else if(index === 0)
            this.add(value)
        else{
            let current = this.head
            while (current.next !== null && index > 1){
                current = current.next
                index -= 1
            }
            current.next = new LinkedListNode(value, current.next)
        }
    }

    contain(value){
        let current = this.head;

        while(current !== null){
            if(current.value === value){
                return true;
            }
            current = current.next;
        }
        return false;
    }

    len() {
        let result=0;
        let current=this.head;

        while (current !== null){
            result+=1;
            current=current.next;
        }

        return result;
    }

    remove(){
        if(this.head===null){
            return undefined;
        }

        const value=this.head.value;

        this.head=this.head.next;

        return value;
    }

    removeAt(index){
        if(this.head===null){
            return undefined;
        }
        else if(index===0||this.head.next===null){
            return this.remove()
        }
        else{
            let current=this.head;
            while(current.next.next!==null && index>1){
                current=current.next;
                index-=1;
            }
            const value=current.next.value;
            current.next=current.next.next;

            return value;
        }
    }
}

File: algorythms/7/linked_list.py
class linkedListNode:
    def __init__(self, value, next):
        self.value=value
        self.next=next

class linkedList:
    head=None

    def add(self, value):
        self.head=linkedListNode(value, self.head)
    
    def addToEnd(self, value):
        current=self.head

        while current.next is not None:
            if current.next is None:
                current=linkedListNode(value, None)

    def deleteOne(self):
        if self.head is None:
            return None
        
        value=self.head.value
        self.head=self.head.next

        return value


    def deleteItem(self, value):
        current=self.head

        if current is None:
            return None

        elif current.next is None:
            return self.deleteOne()
        
        else:
            while current.next is not None and current.value is not value:
                current=current.next
                current.value=value
            
            nextValue=current.next.value
            current.next=current.next.next
        
            return nextValue
        